lib/pure/os.nim

```diff
@@ -2449,25 +2522,31 @@ proc copyFileWithPermissions*(source, dest: string,
   ## `OSError`.
   ##
   ## See also:
+  ## * `CopyFlag enum <#CopyFlag>`_
   ## * `copyFile proc <#copyFile,string,string>`_
   ## * `copyDir proc <#copyDir,string,string>`_
   ## * `tryRemoveFile proc <#tryRemoveFile,string>`_
   ## * `removeFile proc <#removeFile,string>`_
   ## * `moveFile proc <#moveFile,string,string>`_
   ## * `copyDirWithPermissions proc <#copyDirWithPermissions,string,string>`_
-  copyFile(source, dest)
+  copyFile(source, dest, options)
   when not defined(Windows):
-    try:
-      setFilePermissions(dest, getFilePermissions(source))
-    except:
-      if not ignorePermissionErrors:
-        raise
+    if not dest.checkSymlink:
```

Comment:
* should that be `if not source.checkSymlink:` instead?
* what happens if source is a symlink, shouldn't we also set file permissions?  (either copying permissions from symlink or from target of symlink depending on `options`)

Reply:
> should that be `if not source.checkSymlink:` instead?

> On Linux, the permissions of an ordinary symbolic link are not used in any operations; the permissions are always 0777 (read, write, and execute for all user categories), and can't be changed.

(https://man7.org/linux/man-pages/man7/symlink.7.html)

However:
> On most systems, symlink permissions don't matter. When using the symlink, the permissions of the components of symlink's target will be checked. However, on some systems they do matter. MacOS requires read permission on the link for readlink, and NetBSD's symperm mount option forces link permissions checks on read and traversal.

(https://unix.stackexchange.com/a/87202/47186)

I wrote `dest.checkSymlink` because I thought that it's not possible to change the permissions of symlinks on all POSIX OSes, but I was wrong.

So, I don't know how how to properly handle this. Probably, it's worth checking implementation in other languages.

Reply:
OK, I copied the approach from Python's `shutil.copy`. Please, have a look)